### ISSUE TITLE
Consider moving Fastify from devDependencies to peerDependencies

### ISSUE THREAD

**Issue by DevelopmentByDavid** [mercurius-js/mercurius#477]
### TLDR
I believe `fastify` should be moved from devDependencies to peerDependencies

### Problem
When using yarn v2 workspaces, fastify types within mercurius are implicitly evaluated as `any`.  

I can post a reproduction repo that I have set up locally if necessary.

Here is the relevant output from running `yarn workspace <relevant workspace> tsc --traceResolution --noEmit`
```
======== Resolving module 'fastify' from '/home/david/Documents/learn/workspace-mercurius/.yarn/cache/mercurius-npm-7.6.0-7b88275c27-6ab7dcaff6.zip/node_modules/mercurius/index.d.ts'. ========
Explicitly specified module resolution kind: 'NodeJs'.
Loading module 'fastify' from 'node_modules' folder, target file type 'TypeScript'.
Loading module 'fastify' from 'node_modules' folder, target file type 'JavaScript'.
======== Module name 'fastify' was not resolved. ========
```

### Research
- In [mercurius types](https://github.com/mercurius-js/mercurius/blob/master/index.d.ts) file, fastify types are imported. 
- `fastify` is declared as a [devDependency](https://github.com/mercurius-js/mercurius/blob/master/package.json#L39)

### Solution
Patching myself via `.yarnrc.yml`
```yml
packageExtensions:
    'mercurius@*':
        peerDependencies:
            fastify: '*'
```
It'd be nice if the project had `fastify` in peerDependencies since the types exported from mercurius depend on fastify too.  I don't think this change is too much of a problem since mercurius is meant to be used with fastify.  Although, I'm not sure if there are other implications of this change, as I have never maintained an open source package myself.
### Other Things I tried/Notes
- Various typescript settings; I thought this was a typescript module resolution error at first, which it kinda is.
- Various yarn settings before arriving to the solution above
- This problem is only relevant to workspaces.  It is not a problem when the project is not a workspace.
- The problem is also solved by adding `fastify` to the root workspace (not sure why).  Although, I wasn't satisfied with this solution :)

**Comment by romeerez** [comment]
Hi, I'm using mercurius with yarn v2 workspaces too, high five!
But I didn't have any issue with deps, it just works

**Comment by mcollina** [comment]
peerDependencies are weird as they work differently in different version of npm, yarn, pnpm etc. I just avoid them to avoid all sort of problems: fixing it for yarn would _definitely_ break some other use case. Also typescript has its own set of problems.

Thanks, it's good discussion anyway.
